File: gcse-programming-concepts/programming-constructs/condition_controlled_loop.py
def loop_until_correct():
    """An example of a condition controlled loop."""
    name = input("Please enter your name ")
    while name != "Fergus":
       name = input("Please enter your name ")
    print("You guessed my name!")

if __name__ == '__main__':
    loop_until_correct()

File: gcse-programming-concepts/programming-constructs/condition_controlled_loop.cs
using System;

namespace isaac_code_samples
{
    class WhileLoop
    {
        void Main(string[] args){
            /// A name guessing game, user inputs guesses until they get the right answer
            Console.WriteLine("Please enter your name");
            string name = Console.ReadLine();
            while (name != "Fergus"){
                Console.WriteLine("Please enter your name");
                name = Console.ReadLine();
            }
            Console.WriteLine("You guessed my name!");

        }
    }
}